library/std/src/collections/hash/set.rs

```diff
@@ -588,6 +588,13 @@ where
     /// Visits the values representing the intersection,
     /// i.e., the values that are both in `self` and `other`.
     ///
+    /// Note: this operation does not guarantee which collection
+    /// is visited from `self` or `other`. This has consequences
+    /// for values which may be defined as equal by the `Eq` trait
+    /// but which are not physically equivalent (eg. they may have
+    /// fields which differ or do not participate in the definition
+    /// of equivalence).
```

Comment:
This can be phrased more concisely along the lines of 

> When an equal element is present in `self` and `other` then the resulting `Intersection` may yield references to one or the other, which will be visible in properties not covered by `Eq`.

Reply:
Thank you, I've integrated some of this into the PR.